skipped_tests.toml

```diff
@@ -163,6 +163,7 @@ functional = [
 	"tests/pytests/functional/utils/yamllint/test_yamllint.py::test_has_yamllint",
 	"tests/pytests/functional/utils/yamllint/test_yamllint.py::test_input_bytes",
 	"tests/pytests/functional/utils/yamllint/test_yamllint.py::test_version",
+	"tests/pytests/functional/states/pkgrepo/test_debian.py::test_adding_repo_file_arch", # Seems to cause "Socket exception: Connection timed out" on Debian 11
```

Comment:
I'm not sure this is going to work... let me try to elaborate this.

From the Salt Shaker logs for Debian 11, in some executions we see:

```
03:12:18  tests/pytests/functional/states/pkgrepo/test_debian.py::test_adding_repo_file_arch[With(aptsources.sourceslist)] PASSED [ 88%]2024-08-30 04:12:04,970 - ERROR - Socket exception: Connection timed out (110)
```
in some others:
```
03:13:59  tests/pytests/functional/states/pkgrepo/test_debian.py::test_adding_repo_file[With(aptsources.sourceslist)] PASSED [ 88%]2024-09-08 04:13:46,346 - ERROR - Socket exception: Connection timed out (110)
```

Notice the mentioned tests are different from each execution and actually they are set as `PASSED` by pytest.

The error message we see there I think it actually comes from the `terracumber` execution (the CLI that drives the pipeline execution). To me, it sounds like the machine is just rebooted/turnedoff/networkdown around that time. That would explain why the pipeline also produces an error later on when trying to execute the post execution stages for getting the results:

```
03:13:51  + ./terracumber-cli --outputdir /home/jenkins/workspace/manager-salt-shaker-products-testing-debian11-bundle/results --tf susemanager-ci/terracumber_config/tf_files/salt-shaker/Salt-Shaker-Testing-Debian11-Bundle.tf --gitfolder /home/jenkins/workspace/manager-salt-shaker-products-testing-debian11-bundle/results/sumaform --terraform-bin /usr/bin/terraform --logfile /home/jenkins/workspace/manager-salt-shaker-products-testing-debian11-bundle/results/96/testsuite.log --runstep saltshaker_getresults
03:14:06  Fetching files from Salt Shaker node to /home/jenkins/workspace/manager-salt-shaker-products-testing-debian11-bundle/results/96/...
03:14:06  Traceback (most recent call last):
03:14:06    File "./terracumber-cli", line 477, in <module>
03:14:06      main()
03:14:06    File "./terracumber-cli", line 459, in main
03:14:06      results['results'] = get_results_saltshaker(args, tf_vars, config, ctl_creds)
03:14:06    File "./terracumber-cli", line 285, in get_results_saltshaker
03:14:06      cucumber = terracumber.cucumber.Cucumber(ctl_creds, False, 'AutoAddPolicy')
03:14:06    File "/home/jenkins/workspace/manager-salt-shaker-products-testing-debian11-bundle/terracumber/cucumber.py", line 29, in __init__
03:14:06      self.ssh_client.connect(**conn_data)
03:14:06    File "/usr/lib/python3.6/site-packages/paramiko/client.py", line 362, in connect
03:14:06      raise NoValidConnectionsError(errors)
03:14:06  paramiko.ssh_exception.NoValidConnectionsError: [Errno None] Unable to connect to port 22 on 10.146.11.39
```

^^ like the host is down at the point in time.

Another interesting thing I noticed is that, if we trigger the pipeline execution during work hours, then it succeed, apparentely only the automatic midnight execution is the one that failed.

Example:
- Succeeded 30 ago. 2024 13:07 CEST: https://ci.suse.de/user/manager/my-views/view/Salt%20Shaker/job/manager-salt-shaker-products-testing-debian11-bundle/86/
- Failed 31 ago. 2024 2:30 CEST: https://ci.suse.de/user/manager/my-views/view/Salt%20Shaker/job/manager-salt-shaker-products-testing-debian11-bundle/87/

I also strangely see that, on the nightly executions, there seems to be an stuck time (~15min) here, just before we see the error:

```
02:55:12  tests/pytests/functional/states/pkgrepo/test_centos.py::test_copr_pkgrepo_with_comments SKIPPED [ 88%]
02:55:12  tests/pytests/functional/states/pkgrepo/test_debian.py::test_adding_repo_file[With(aptsources.sourceslist)] PASSED [ 88%]
02:55:12  tests/pytests/functional/states/pkgrepo/test_debian.py::test_adding_repo_file_arch[With(aptsources.sourceslist)] SUBPASS [ 88%]
03:12:18  tests/pytests/functional/states/pkgrepo/test_debian.py::test_adding_repo_file_arch[With(aptsources.sourceslist)] PASSED [ 88%]2024-08-30 04:12:04,970 - ERROR - Socket exception: Connection timed out (110)
03:12:18  
03:12:18  tests/pytests/functional/states/pkgrepo/test_debian.py::test_adding_repo_file_cdrom[With(aptsources.sourceslist)] PASSED [ 88%]
```

When pipeline is triggered manually during work hours, then pipeline is successful and we don't see this stuck period happening:
```
12:58:37  tests/pytests/functional/states/pkgrepo/test_centos.py::test_pkgrepo_managed_absent SKIPPED [ 87%]
12:58:37  tests/pytests/functional/states/pkgrepo/test_centos.py::test_pkgrepo_with_comments SKIPPED [ 88%]
12:58:37  tests/pytests/functional/states/pkgrepo/test_centos.py::test_copr_pkgrepo_with_comments SKIPPED [ 88%]
12:58:37  tests/pytests/functional/states/pkgrepo/test_debian.py::test_adding_repo_file[With(aptsources.sourceslist)] PASSED [ 88%]
12:58:37  tests/pytests/functional/states/pkgrepo/test_debian.py::test_adding_repo_file_arch[With(aptsources.sourceslist)] SUBPASS [ 88%]
12:59:58  tests/pytests/functional/states/pkgrepo/test_debian.py::test_adding_repo_file_arch[With(aptsources.sourceslist)] PASSED [ 88%]
12:59:58  tests/pytests/functional/states/pkgrepo/test_debian.py::test_adding_repo_file_cdrom[With(aptsources.sourceslist)] PASSED [ 88%]
12:59:58  tests/pytests/functional/states/pkgrepo/test_debian.py::test_pkgrepo_managed_absent[With(aptsources.sourceslist)] SKIPPED [ 88%]
12:59:58  tests/pytests/functional/states/pkgrepo/test_debian.py::test_managed_multiple_comps[With(aptsources.sourceslist)] SKIPPED [ 88%]
12:59:58  tests/pytests/functional/states/pkgrepo/test_debian.py::test_pkgrepo_with_architectures[With(aptsources.sourceslist)] SUBPASS [ 88%]
12:59:58  tests/pytests/functional/states/pkgrepo/test_debian.py::test_pkgrepo_with_architectures[With(aptsources.sourceslist)] SUBPASS [ 88%]
12:59:58  tests/pytests/functional/states/pkgrepo/test_debian.py::test_pkgrepo_with_architectures[With(aptsources.sourceslist)] SUBPASS [ 88%]
```

So, I would maybe try to figure out why this failure only happens at night and pipeline is successful  when triggered manually during working hours.

Reply:
So it's true that manual executions pass while the automatic ones fail, the reason for this is not known. But:

> To me, it sounds like the machine is just rebooted/turnedoff/networkdown around that time.

I think we can rule this out as of right now. 

If you try to SSH into the salt-shaker Debian 11 VM, you'll see that the VM uses IPv6 only. For some reason, the test (or one of the tests) seems to remove the IPv4 config and the default routes, which means the VM only responds to IPv6 from the VM host. The jenkins pipeline cannot connect to the VM, which is why it fails to download the test results (and the Jenkins logs), and times out after certain time.

The only way to re-connect to the machine is to reprovision it. You can check the actual VM, since it's in the failed state:

![image](https://github.com/user-attachments/assets/138871f9-6584-4be3-aac6-0748288027fe)

You are correct that this might not be the actual test that causes the failure (or, it might be multiple tests that cause the failure). Both Yeray and I have done some basic testing with skiplist entries and this seems to me like the most likely candidate. However, it is totally possible we'll have to re-adjust the tests, since it seems like we cannot replicate the fails with manual runs.

Why manual runs pass and automated runs fail is indeed unknown. But, before spending another days on diagnosing this strange issue, I think it's more prudent to try an easier fix, even if it might require further tweaking (or even if we'll revert it later on).

Reply:
Ok, fair enough. Let's give it a try to this, so we can have the green here and we will revisit this anyways once we tackle this other card: https://github.com/SUSE/spacewalk/issues/23536